### ISSUE TITLE
InvalidRefException when processing paths result in cartographer

### ISSUE THREAD

**Issue by pkocandr** [Commonjava/atlas#50]
```
    20:13:32.896 [XNIO-1 task-4] ERROR io.undertow.request - UT005023: Exception handling request to /api/depgraph/graph/paths
    org.jboss.resteasy.spi.UnhandledException: org.commonjava.maven.atlas.ident.ref.InvalidRefException: ProjectId must contain non-empty groupId AND artifactId. (Given: 'null':'null', db-node=3281)
            at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:76) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:212) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:149) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:372) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:179) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:220) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[jboss-servlet-api_3.1_spec-1.0.0.Final.jar:1.0.0.Final]
            at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:130) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at org.commonjava.aprox.bind.jaxrs.ResourceManagementFilter.doFilter(ResourceManagementFilter.java:60) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
            at org.commonjava.aprox.bind.jaxrs.ResourceManagementFilter$Proxy$_$$_WeldClientProxy.doFilter(Unknown Source) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
            at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:132) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:85) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:61) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:56) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:45) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:63) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:58) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:70) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.security.handlers.SecurityInitialHandler.handleRequest(SecurityInitialHandler.java:76) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
            at org.commonjava.aprox.bind.jaxrs.HeaderDebugger.handleRequest(HeaderDebugger.java:93) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
            at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:261) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:247) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:76) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:166) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.server.Connectors.executeRootHandler(Connectors.java:197) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
            at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:764) [undertow-core-1.1.2.Final.jar:1.1.2.Final]
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_60]
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_60]
            at java.lang.Thread.run(Thread.java:745) [na:1.8.0_60]
    Caused by: org.commonjava.maven.atlas.ident.ref.InvalidRefException: ProjectId must contain non-empty groupId AND artifactId. (Given: 'null':'null', db-node=3281)
            at org.commonjava.maven.atlas.graph.spi.neo4j.model.NeoProjectRef.<init>(NeoProjectRef.java:91) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
            at org.commonjava.maven.atlas.graph.spi.neo4j.model.NeoProjectVersionRef.<init>(NeoProjectVersionRef.java:51) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
            at org.commonjava.maven.atlas.graph.spi.neo4j.model.NeoArtifactRef.<init>(NeoArtifactRef.java:84) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
            at org.commonjava.maven.atlas.graph.spi.neo4j.model.NeoDependencyRelationship.getTarget(NeoDependencyRelationship.java:107) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
            at org.commonjava.maven.atlas.graph.spi.neo4j.model.NeoDependencyRelationship.getTarget(NeoDependencyRelationship.java:40) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
            at org.commonjava.maven.atlas.graph.spi.neo4j.model.AbstractNeoProjectRelationship.equals(AbstractNeoProjectRelationship.java:199) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
            at org.commonjava.maven.atlas.graph.spi.neo4j.model.NeoDependencyRelationship.equals(NeoDependencyRelationship.java:79) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
            at java.util.HashMap.getNode(HashMap.java:571) ~[na:1.8.0_60]
            at java.util.HashMap.containsKey(HashMap.java:595) ~[na:1.8.0_60]
            at java.util.HashSet.contains(HashSet.java:203) ~[na:1.8.0_60]
            at org.commonjava.cartographer.INTERNAL.ops.GraphOpsImpl.lambda$getPaths$6(GraphOpsImpl.java:126) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
            at org.commonjava.cartographer.graph.GraphResolver.resolveAndExtractMultiGraph(GraphResolver.java:184) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
            at org.commonjava.cartographer.INTERNAL.ops.GraphOpsImpl.getPaths(GraphOpsImpl.java:139) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
            at org.commonjava.aprox.depgraph.rest.GraphController.getPaths(GraphController.java:85) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
            at org.commonjava.aprox.depgraph.jaxrs.GraphResource.getPaths(GraphResource.java:56) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
            at org.commonjava.aprox.depgraph.jaxrs.GraphResource$Proxy$_$$_WeldClientProxy.getPaths(Unknown Source) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
            at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_60]
            at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_60]
            at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_60]
            at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_60]
            at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:137) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:296) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:250) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:237) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:356) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
            ... 34 common frames omitted
```

This was revealed in AProx 0.25.2-SNAPSHOT (atlas 0.15.3-SNAPSHOT) after I fixed an NPE in RelationshipGraphFactory$ConnectionCache.startCloseTimer(RelationshipGraphFactory.java:243) when called with connection in the cache set to null, i.e. the cache and the connection was already closed. Maybe the closed graph connection is the problem.

**Comment by pkocandr** [comment]
Another neo4j exception on the same place in cartographer's GraphOpsImpl:
```
02:35:53.720 [XNIO-1 task-6] ERROR io.undertow.request - UT005023: Exception handling request to /api/depgraph/graph/paths
org.jboss.resteasy.spi.UnhandledException: org.neo4j.graphdb.NotFoundException: Relationship[252007] exist but relationship type[3] not found.
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:76) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:212) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:149) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:372) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:179) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:220) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[jboss-servlet-api_3.1_spec-1.0.0.Final.jar:1.0.0.Final]
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:130) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at org.commonjava.aprox.bind.jaxrs.ResourceManagementFilter.doFilter(ResourceManagementFilter.java:60) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
	at org.commonjava.aprox.bind.jaxrs.ResourceManagementFilter$Proxy$_$$_WeldClientProxy.doFilter(Unknown Source) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:60) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:132) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:85) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:61) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:56) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:45) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:63) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:58) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:70) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.security.handlers.SecurityInitialHandler.handleRequest(SecurityInitialHandler.java:76) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
	at org.commonjava.aprox.bind.jaxrs.HeaderDebugger.handleRequest(HeaderDebugger.java:93) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:261) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:247) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:76) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:166) ~[undertow-servlet-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:197) ~[undertow-core-1.1.2.Final.jar:1.1.2.Final]
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:764) [undertow-core-1.1.2.Final.jar:1.1.2.Final]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_60]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_60]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_60]
Caused by: org.neo4j.graphdb.NotFoundException: Relationship[252007] exist but relationship type[3] not found.
	at org.neo4j.kernel.impl.core.NodeManager.getRelationshipForProxy(NodeManager.java:681) ~[neo4j-kernel-1.9.6.jar:1.9.6]
	at org.neo4j.kernel.InternalAbstractGraphDatabase$7.lookupRelationship(InternalAbstractGraphDatabase.java:747) ~[neo4j-kernel-1.9.6.jar:1.9.6]
	at org.neo4j.kernel.impl.core.RelationshipProxy.getStartNode(RelationshipProxy.java:88) ~[neo4j-kernel-1.9.6.jar:1.9.6]
	at org.commonjava.maven.atlas.graph.spi.neo4j.model.AbstractNeoProjectRelationship.getDeclaring(AbstractNeoProjectRelationship.java:114) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
	at org.commonjava.maven.atlas.graph.spi.neo4j.model.AbstractNeoProjectRelationship.hashCode(AbstractNeoProjectRelationship.java:166) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
	at org.commonjava.maven.atlas.graph.spi.neo4j.model.NeoDependencyRelationship.hashCode(NeoDependencyRelationship.java:67) ~[atlas-driver-neo4j-embedded-0.15.3-SNAPSHOT.jar:na]
	at java.util.HashMap.hash(HashMap.java:338) ~[na:1.8.0_60]
	at java.util.HashMap.containsKey(HashMap.java:595) ~[na:1.8.0_60]
	at java.util.HashSet.contains(HashSet.java:203) ~[na:1.8.0_60]
	at org.commonjava.cartographer.INTERNAL.ops.GraphOpsImpl.lambda$getPaths$6(GraphOpsImpl.java:126) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
	at org.commonjava.cartographer.graph.GraphResolver.resolveAndExtractMultiGraph(GraphResolver.java:184) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
	at org.commonjava.cartographer.INTERNAL.ops.GraphOpsImpl.getPaths(GraphOpsImpl.java:139) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
	at org.commonjava.aprox.depgraph.rest.GraphController.getPaths(GraphController.java:85) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
	at org.commonjava.aprox.depgraph.jaxrs.GraphResource.getPaths(GraphResource.java:56) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
	at org.commonjava.aprox.depgraph.jaxrs.GraphResource$Proxy$_$$_WeldClientProxy.getPaths(Unknown Source) ~[aprox-embedder-savant-0.25.2-SNAPSHOT.jar:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_60]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_60]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_60]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_60]
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:137) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:296) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:250) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:237) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:356) ~[resteasy-jaxrs-3.0.9.Final.jar:na]
	... 34 common frames omitted
```

**Comment by pkocandr** [comment]
This seems to be fixed by https://github.com/Commonjava/cartographer/pull/91. I'm running more tests.

**Comment by pkocandr** [comment]
Ah, when the connection is not closed it raises the second exception (Relationship[403608] exist but relationship type[3] not found).

**Comment by pkocandr** [comment]
I've cleaned the workspaces and am building the graphs again to see if it was only a problem in the neo4j DB.

**Comment by pkocandr** [comment]
InvalidRefException is fixed, the NotFoundException was filed as #55.
